### PR TITLE
Add FLE 2 API to AutoEncryptionOpts

### PR DESCRIPTION
**Pull request by kkloberdanz**
CXX-2500

## Driver changes
Add the new driver API options: AutoEncryptionOpts.encryptedFieldsMap and AutoEncryptionOpts.bypassQueryAnalysis.

Pass AutoEncryptionOpts.encryptedFieldsMap to libmongocrypt with mongoc_auto_encryption_opts_set_bypass_query_analysis.

## Test changes
Add encrypted_fields to the CSFLE specification test format.

Sync the added CSFLE specification tests.

### CODE REVIEW

**Review by kevinAlbs — APPROVED** [review]
Looks good.

Suggest updating the PR description. Some of the description does not apply to the C++ driver. The C++ driver uses the C driver for some of the functionality:

> Pass AutoEncryptionOpts.encryptedFieldsMap to libmongocrypt with ~~mongocrypt_setopt_encrypted_field_config_map~~ mongoc_auto_encryption_opts_set_bypass_query_analysis.

The following step is done by the C driver be removed:

> If AutoEncryptionOpts.bypassQueryAnalysis is true, call the libmongocrypt function mongocrypt_setopt_bypass_query_analysis~~.

File: src/mongocxx/test/spec/util.cpp [hunk 1]
```diff
@@ -222,11 +222,16 @@ void set_up_collection(const client& client,
     auto db = client[test[database_name].get_string().value];
     db.drop();
 
+    bsoncxx::builder::basic::document opts;
+    if (test["encrypted_fields"]) {
+        opts.append(kvp("encryptedFields", test["encrypted_fields"].get_document().value));
+    }
```
Comment (eramongodb) [inline src/mongocxx/test/spec/util.cpp:228]:
```suggestion
    bsoncxx::builder::basic::document opts;
    if (const auto fields = test["encrypted_fields"]) {
        opts.append(kvp("encryptedFields", fields.get_document().value));
    }
```
Avoid double-lookup.

File: src/mongocxx/test/spec/client_side_encryption.cpp [hunk 2]
```diff
@@ -61,6 +61,11 @@ void add_auto_encryption_opts(document::view test, options::client* client_opts)
                 test_encrypt_opts["bypassAutoEncryption"].get_bool().value);
         }
 
+        if (test_encrypt_opts["bypassQueryAnalysis"]) {
+            auto_encrypt_opts.bypass_query_analysis(
+                test_encrypt_opts["bypassQueryAnalysis"].get_bool().value);
+        }
```
Comment (eramongodb) [inline src/mongocxx/test/spec/client_side_encryption.cpp:67]:
```suggestion
        if (const auto bqa = test_encrypt_opts["bypassQueryAnalysis"]) {
            auto_encrypt_opts.bypass_query_analysis(bqa.get_bool().value);
        }
```
Avoid double-lookup.

**Review by eramongodb — APPROVED** [review]
Minor suggestions; otherwise, LGTM.

File: src/mongocxx/test/spec/util.cpp [hunk 3]
```diff
@@ -222,11 +222,17 @@ void set_up_collection(const client& client,
     auto db = client[test[database_name].get_string().value];
     db.drop();
 
+    bsoncxx::builder::basic::document opts;
+    bsoncxx::builder::basic::document opts;
```
Comment (vector-of-bool) [inline src/mongocxx/test/spec/util.cpp:226]: It looks like a goofed merge caused this variable declaration to be duplicated.

**Review by vector-of-bool — APPROVED** [review]
LGTM, with one goof.

File: src/mongocxx/test/spec/util.cpp (hunk 3 above)
Comment (kkloberdanz) [reply src/mongocxx/test/spec/util.cpp]: Haha, nice! Fixed 😄

**Pull request merged**